include/cuco/detail/trie/bit_vector/bit_vector_ref.inl

```diff
@@ -0,0 +1,238 @@
+#include <cuco/detail/trie/bit_vector/bit_vector.cuh>
+
+namespace cuco {
+namespace experimental {
+
+template <typename StorageRef, typename... Operators>
+__host__ __device__ constexpr bit_vector_ref<StorageRef, Operators...>::bit_vector_ref(
+  StorageRef words_ref,
+  StorageRef ranks_ref,
+  StorageRef selects_ref,
+  StorageRef ranks0_ref,
+  StorageRef selects0_ref) noexcept
+  : words_ref_{words_ref},
+    ranks_ref_{ranks_ref},
+    selects_ref_{selects_ref},
+    ranks0_ref_{ranks0_ref},
+    selects0_ref_{selects0_ref}
+{
+}
+
+namespace detail {
+
+template <typename StorageRef, typename... Operators>
+class operator_impl<op::bv_read_tag, bit_vector_ref<StorageRef, Operators...>> {
+  using ref_type  = bit_vector_ref<StorageRef, Operators...>;  ///< Bitvector ref type
+  using size_type = typename StorageRef::size_type;            ///< Size type
+  using slot_type = typename StorageRef::value_type;           ///< Slot type
+
+  const size_type bits_per_word   = sizeof(slot_type) * 8;
+  const size_type words_per_block = 4;  //< This should match the defintion in bit_vector
+
+ public:
+  /**
+   * @brief Access value of a single bit
+   *
+   * @param key Position of bit
+   *
+   * @return Value of bit at position specified by key
+   */
+  [[nodiscard]] __device__ bool get(size_type key) const noexcept
+  {
+    auto const& ref_ = static_cast<ref_type const&>(*this);
+    return (ref_.words_ref_[key / bits_per_word][0] >> (key % bits_per_word)) & 1UL;
+  }
+
+  /**
+   * @brief Find position of first set bit starting from a given position (inclusive)
+   *
+   * @param key Position of starting bit
+   *
+   * @return Index of next set bit
+   */
+  [[nodiscard]] __device__ size_type find_next_set(size_type key) const noexcept
+  {
+    auto const& ref_  = static_cast<ref_type const&>(*this);
+    size_type word_id = key / bits_per_word;
+    size_type bit_id  = key % bits_per_word;
```

Comment:
Declaring `bits_per_word`as `constexpr` should speed up these otherwise expensive operations `/` and `%` considerably.